### DISCUSSION TITLE
Default timeout for the HTTP client

### DISCUSSION THREAD

**Discussion by mnapoli** [laravel/framework#36710] [Ideas]
Hi,

Would it make sense to allow changing the default timeout for the HTTP client?

When running Laravel on Lambda, there is a hard limit of 30 seconds, after which the PHP request is completely stopped.

Being able to set the default timeout on everything at ~10 or 20 seconds would allow having a more precise error instead.

Yes, it is possible to define the timeout on every request made. But sometimes the request is made in a package, it's not possible to change every HTTP request in the app.

I have not found a way to set that up so far, AFAICT it requires changing the code in Laravel.

**Comment by vdauchy** [comment]
Hi,

I just did a PR on same need: https://github.com/laravel/framework/pull/36753

**Reply by mnapoli** [reply]
Awesome!
